src/main/java/org/akaza/openclinica/service/ValidateServiceImpl.java

```diff
@@ -251,6 +256,34 @@ public boolean isUserHas_DM_MON_RoleInStudy(List<StudyUserRoleBean> userRoles, S
         }
         return false;
     }
+
+    public boolean isUserHasAccessToStudyOrSiteForStudy(List<StudyUserRoleBean> userRoles, String studyOid){
+        Study study = studyDao.findPublicStudy(studyOid);
+        Study tempPublicStudy = null;
+        for (StudyUserRoleBean userRole : userRoles) {
+            tempPublicStudy = study;
+            if(Role.siteRoleMap.containsKey(userRole.getRole().getId()))
+                tempPublicStudy = studyDao.getPublicStudy(userRole.getStudyId());
+            if (study.getStudyId() == tempPublicStudy.getStudyId() || (tempPublicStudy.isSite() && study.getStudyId() == tempPublicStudy.getStudy().getStudyId())) {
```

Comment:
I don't see the logic for check userRole.getStudyId in the case of study-level role.